ebi_eva_common_pyutils/metadata_utils.py

```diff
@@ -101,31 +105,152 @@ def resolve_existing_variant_warehouse_db_name(metadata_connection_handle, assem
+
+
+def insert_new_assembly_and_taxonomy(metadata_connection_handle, assembly_accession, taxonomy_id, eva_species_name=None,
+                                     in_accessioning=True):
+    """
+    This script adds new assemblies and taxonomies to EVAPRO.
+    You can also add the assembly with a different taxonomy if you provide the
+    taxonomy parameters. Example taxonomy page:
+    https://www.ebi.ac.uk/ena/data/view/Taxon:9031
+
+    :param assembly_accession: Assembly accession (Example: GCA_000002315.3)
+    :param metadata_connection_handle: Metadata DB connection
+    :param taxonomy_id: Taxonomy id (Example: 9031)
+    :param eva_species_name: EVA species name (Example: chicken).
+        Not required if the taxonomy exists or ENA has a common name available.
+    :param in_accessioning: Flag that this assembly is in the accessioning data store.
+    """
+    assembly_name = get_ncbi_an_assembly_name_from_term(assembly_accession)
+
+    # check if assembly is already in EVAPRO, adding it if not
+    assembly_set_id = get_assembly_set_from_metadata(metadata_connection_handle, taxonomy_id, assembly_accession)
+    if assembly_set_id is None:
+        ensure_taxonomy_is_in_evapro(metadata_connection_handle, taxonomy_id, eva_species_name)
+        assembly_code = get_assembly_code(metadata_connection_handle, assembly_accession)
+        insert_assembly_in_evapro(metadata_connection_handle, taxonomy_id, assembly_accession, assembly_name, assembly_code)
+
+    update_accessioning_status(metadata_connection_handle, assembly_accession, in_accessioning)
+    metadata_connection_handle.commit()
+
+
+def ensure_taxonomy_is_in_evapro(metadata_connection_handle, taxonomy, eva_species_name=None):
+    if is_taxonomy_in_evapro(metadata_connection_handle, taxonomy):
+        logger.debug('Taxonomy {} is already in the database'.format(taxonomy))
+    else:
+        logger.info("Taxonomy {} not present in EVAPRO. Adding taxonomy ...".format(taxonomy))
+        scientific_name, common_name = get_scientific_name_and_common_name(taxonomy)
+        taxonomy_code = build_taxonomy_code(scientific_name)
+        eva_species_name = eva_species_name if eva_species_name is not None else common_name
+        if eva_species_name is None:
+            raise ValueError(
+                'The taxonomy in ENA doesn\'t include a common name. '
+                'Please specify the EVA name for the species "{}"'.format(scientific_name))
+        insert_taxonomy(metadata_connection_handle, taxonomy, scientific_name, common_name, taxonomy_code, eva_species_name)
+
+
+def insert_assembly_in_evapro(metadata_connection_handle, taxonomy_id, assembly_accession, assembly_name, assembly_code):
+    cur = metadata_connection_handle.cursor()
+    cur.execute('INSERT INTO evapro.assembly_set(taxonomy_id, assembly_name, assembly_code) VALUES (%s, %s, %s)',
+                (taxonomy_id, assembly_name, assembly_code))
```

Comment:
At some point we should consider cleaning this up to use [our utility methods](https://github.com/tcezard/eva-common-pyutils/blob/60f151c0fdf3121fc05d40df64d708e7e4dfe66e/ebi_eva_common_pyutils/pg_utils.py)... I think this is only like this because it's ported from some even older scripts...

Reply:
Not a big fan of these utility method, they force us to create the string of the query where the ones using the cursor directly allow for automatic type casting so we don't need to add quotes. Again I think I would prefer switching to an ORM.